### PR TITLE
Add support for `kickstart` to `mrack` plugin

### PR DESCRIPTION
**Pull request by skycastlelily**
Pull Request Checklist

* [x] implement the feature
* [x] update the specification
* [x] modify the json schema
* [x] mention the version
* [x] include a release note

### CODE REVIEW

File: tmt/utils/__init__.py [hunk 1]
```diff
@@ -78,6 +78,34 @@
     from tmt._compat.typing import Self, TypeAlias
 
 
+def _normalize_user_data(
```
Comment (happz) [inline tmt/utils/__init__.py:81]: By moving into `tmt.utils`:

* the function is no longer "private", used by `artemis` plugin and no one else, therefore it should no longer start with an underscore,
* "user data" is a concept understood by Artemis and therefore by `artemis`, but it does not mean much to the rest of the code base. I would suggest finding a different name and adding a docstring. It was clear what was happening when it was in `tmt.steps.provision.artemis`, but what is `tmt.utils._normalize_user_data` doing? tmt does not accept any "user data"...

**Comment by skycastlelily** [comment]
> I would suggest finding a different name and adding a docstring

Thanks for your mentor,How about"normalize_dict"? :)

**Comment by happz** [comment]
> I would suggest finding a different name and adding a docstring
> Thanks for your mentor,How about"normalize_dict"? :)
> […](#)

Still way too generic, I'm afraid, it can mean anything, any input, any type of keys and values, who knows, it's very open. We need a name that reflects it is normalizing `key=value` inputs, and strings on both sides, sometimes the value would be optional...

**Comment by skycastlelily** [comment]
Updated^^

**Comment by happz** [comment]
@skycastlelily I tried to polish it a bit, I was not very happy about `beaker` field, that leaks Mrack's weird API into our code, so `to_mrack()` now exists. Check https://github.com/teemtee/tmt/pull/3064/commits/466b5686d9d73b3e8c1b591289be8ebc372983e1, WDYT?

I also change the name of that normalization callback, to `normalize_string_dict`, and I'm pretty happy about it, because it's what it does, normalizes two different forms of key/value inputs into a `dict[str, str]` :)

**Comment by skycastlelily** [comment]
> WDYT?

I  really think it's brilliant, thanks do me the favor to co-author this
merge request and make the code look pretty while I'm on vacation^^

**Review by therazix — APPROVED** [review]

**Review by psss — APPROVED** [review]
Nice, thanks for implementing this!

**Comment by psss** [comment]
Updated spec and added a release note in 5220d375.

**Comment by skycastlelily** [comment]
Thanks:)

**Pull request merged**